### ISSUE TITLE
[Bug] Player doesnt immediately stop at end destination when not using navigation

### ISSUE THREAD

**Issue by Linnnnnna** [StormyBytes/mindustry-client#5]
https://user-images.githubusercontent.com/63439268/180583998-464ac40e-8fb5-499c-ac86-d6705fdfa8d2.mp4

**Comment by sbxte** [comment]
hmm, I cant seem to play the video

**Comment by Linnnnnna** [comment]
> hmm, I cant seem to play the video

<s>try download it</s>

**Comment by sbxte** [comment]
Oh yeah okay, I see what you mean now

**Comment by sbxte** [comment]
why dont you use navpath anyways? Zxtej's optimized pathing is much smoother

**Comment by Linnnnnna** [comment]
> why dont you use navpath anyways? Zxtej's optimized pathing is much smoother

<s>idk</s>

**Comment by Linnnnnna** [comment]
<s>are you sure about mining thorium</s>
![图片](https://user-images.githubusercontent.com/63439268/180585304-8f7a31bf-1505-49a9-8fc5-8a50493e8c18.png)

**Comment by sbxte** [comment]
Removed in a90633267
Thats a setting anyways, you can change it. Its default mine path args under the misc category

**Comment by Lin-cat** [comment]
what about use white dotted line to draw line

**Comment by Linnnnnna** [comment]
https://user-images.githubusercontent.com/63439268/180588467-ffe04aed-bcd7-4300-955a-5495a8859516.mp4

https://user-images.githubusercontent.com/63439268/180588496-5d735a84-3a9b-40fb-9dbf-141e43782cfe.mp4

**Comment by Linnnnnna** [comment]
![B@8}0FU%LXR%P5A2LIJ@K(3](https://user-images.githubusercontent.com/63439268/180632276-72c4d4bb-4e50-4f3f-99dd-c99a16c1c5c3.png)
![3~%))9JEK1USXSCD$ 8DHH0](https://user-images.githubusercontent.com/63439268/180632278-48460a61-55ac-4e5d-a6c6-78dfc5add399.png)

when press K

**Comment by sbxte** [comment]
Are you on latest version?

**Comment by Linnnnnna** [comment]
> Are you on latest version?
![@@DVBTVI(HT8(0HW{($ 5TD](https://user-images.githubusercontent.com/63439268/180632372-a1cf7049-4562-4378-aaa0-8d30847cf101.png)

**Comment by Linnnnnna** [comment]
> Are you on latest version?

same

![QQ浏览器截图20220724145443](https://user-images.githubusercontent.com/63439268/180636090-8bd36a2e-e906-403a-a847-f8d718172f31.png)

**Comment by Linnnnnna** [comment]
![QQ浏览器截图20220724155224](https://user-images.githubusercontent.com/63439268/180637715-437402ca-8afa-4e9f-8fcf-3b0d1b2e76e8.png)

**Comment by sbxte** [comment]
> ![QQ浏览器截图20220724155224](https://user-images.githubusercontent.com/63439268/180637715-437402ca-8afa-4e9f-8fcf-3b0d1b2e76e8.png)

fixed 2834748f7

**Comment by sbxte** [comment]
> 1.mp4 
>  2.mp4

Regarding this, does zero drift setting affect it? 
Another note, it might take me a while to fix as its hard for me to reproduce the issue and is rather inconsistent on my end.

**Comment by Linnnnnna** [comment]
> > 1.mp4
> > 2.mp4
> 
> Regarding this, does zero drift setting affect it? Another note, it might take me a while to fix as its hard for me to reproduce the issue and is rather inconsistent on my end.

<s>idk, the chance too low</s>

**Comment by Linnnnnna** [comment]
<s>how to delete a default branch</s>

**Comment by sbxte** [comment]
https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/changing-the-default-branch
default branches cannot be deleted, you can only change what the default branch is to another branch.

**Comment by Linnnnnna** [comment]
player.sendMessage(Core.bundle.format("client.command.xx"))  will this work?

**Comment by sbxte** [comment]
depends is Vars.player imported? is Core imported?

**Comment by Linnnnnna** [comment]
> depends is Vars.player imported? is Core imported?

![SQ)% `_X{ETT QGX%S5@170](https://user-images.githubusercontent.com/63439268/180783360-55771d4e-0a28-4b0c-b0e6-82eafedd6f36.png)

**Comment by sbxte** [comment]
oh well then in that case yes, because player and Core are imported in that file

**Comment by Linnnnnna** [comment]
![22PE%HTA%5 0}MT97X8O5LC](https://user-images.githubusercontent.com/63439268/181455515-774ed53a-4578-4092-aa6c-3007168ac896.png)
client.path.miner.tobuild's text （cooper、lead、titanium、thorium）
<s>what about remove thorium</s>

**Comment by sbxte** [comment]
Remove it from bundles then

**Comment by Linnnnnna** [comment]
![S)$SIK}W5U}WG@ UCU3HEEP](https://user-images.githubusercontent.com/63439268/181456793-d78e39f6-ec0d-48f4-aff2-4a7f019c4229.png)

**Comment by sbxte** [comment]
> ![B@8}0FU%LXR%P5A2LIJ@K(3](https://user-images.githubusercontent.com/63439268/180632276-72c4d4bb-4e50-4f3f-99dd-c99a16c1c5c3.png) ![3~%))9JEK1USXSCD$ 8DHH0](https://user-images.githubusercontent.com/63439268/180632278-48460a61-55ac-4e5d-a6c6-78dfc5add399.png)
> 
> when press K

Okay I found the issue. You had to input it based on the localized name. Basically the item's name in what bundle language you're using.

**Comment by Linnnnnna** [comment]
https://user-images.githubusercontent.com/63439268/181866452-213893f2-3305-4c41-bec7-e7b676df3e25.mp4

quickly disappear and appear
is this my issues?

**Comment by Linnnnnna** [comment]
https://user-images.githubusercontent.com/63439268/181867357-622dc5b6-166a-44d8-a606-2cb355fe4c2e.mp4

zero drift doen't work when use shift+click assist player

**Comment by Linnnnnna** [comment]
and your client can't play campaign

**Comment by Linnnnnna** [comment]
![@BV9)$K0`(OTRYS6F}8MB08](https://user-images.githubusercontent.com/63439268/182076180-54767d54-7f50-4eec-96d2-674e821f8e6a.png)

```java
Version: pre-alpha steam build 135
(Flarogus Client Version: 130, Jul. 30, 2022 Unstable)
Source: stormybytes/mindustry-client-builds
OS: Windows 7 x64 (amd64)
Java Version: 16.0.1
Mods: none (vanilla)

java.lang.NullPointerException: Cannot invoke "arc.struct.Seq.add(Object)" because "this.saves" is null
	at mindustry.game.Saves.saveSector(Saves.java:152)
	at mindustry.core.Control.lambda$playSector$28(Control.java:424)
	at mindustry.core.UI.lambda$loadAnd$10(UI.java:259)
	at arc.util.Timer$1.run(Timer.java:88)
	at arc.util.TaskQueue.run(TaskQueue.java:17)
	at arc.backend.sdl.SdlApplication.loop(SdlApplication.java:156)
	at arc.backend.sdl.SdlApplication.<init>(SdlApplication.java:45)
	at mindustry.desktop.DesktopLauncher.main(DesktopLauncher.java:97)
```

**Comment by Linnnnnna** [comment]
![O 8SQS@ 9ZBA4C`HT~KS 4B](https://user-images.githubusercontent.com/63439268/182560799-e95671d7-fdb2-424d-bb87-6a7150cd7bf2.png)

![T` PT533G$} JF@N`2HYU0M](https://user-images.githubusercontent.com/63439268/182560821-4857b633-f2c8-409f-aa96-b3aba5998de2.png)

@StormyBytes, can make it become disable ?

**Comment by Linnnnnna** [comment]
![QOO{)B(6BRKMK}JE SK%5M](https://user-images.githubusercontent.com/63439268/182564421-505193ff-ce06-4cea-bdbf-b77b70e6ad52.png)

**Comment by sbxte** [comment]
> ![O 8SQS@ 9ZBA4C`HT~KS 4B](https://user-images.githubusercontent.com/63439268/182560799-e95671d7-fdb2-424d-bb87-6a7150cd7bf2.png)
> 
> ![T PT533G$} JF@N2HYU0M](https://user-images.githubusercontent.com/63439268/182560821-4857b633-f2c8-409f-aa96-b3aba5998de2.png)
> 
> @StormyBytes, can make it become disable ?

just have it be empty and it wont send anything.


> ![QOO{)B(6BRKMK}JE SK%5M](https://user-images.githubusercontent.com/63439268/182564421-505193ff-ce06-4cea-bdbf-b77b70e6ad52.png)

I am aware of this issue but thanks for pointing it out anyways. Its an issue on arc's end so I might have to come up with some hacky fix. 



> ![@BV9)$K0`(OTRYS6F}8MB08](https://user-images.githubusercontent.com/63439268/182076180-54767d54-7f50-4eec-96d2-674e821f8e6a.png)
> 
> ```java
> Version: pre-alpha steam build 135
> (Flarogus Client Version: 130, Jul. 30, 2022 Unstable)
> Source: stormybytes/mindustry-client-builds
> OS: Windows 7 x64 (amd64)
> Java Version: 16.0.1
> Mods: none (vanilla)
> 
> java.lang.NullPointerException: Cannot invoke "arc.struct.Seq.add(Object)" because "this.saves" is null
> 	at mindustry.game.Saves.saveSector(Saves.java:152)
> 	at mindustry.core.Control.lambda$playSector$28(Control.java:424)
> 	at mindustry.core.UI.lambda$loadAnd$10(UI.java:259)
> 	at arc.util.Timer$1.run(Timer.java:88)
> 	at arc.util.TaskQueue.run(TaskQueue.java:17)
> 	at arc.backend.sdl.SdlApplication.loop(SdlApplication.java:156)
> 	at arc.backend.sdl.SdlApplication.<init>(SdlApplication.java:45)
> 	at mindustry.desktop.DesktopLauncher.main(DesktopLauncher.java:97)
> ```

This is fixed after the erekir merge.


> bandicam.2022-07-30.10-46-03-684.mp4 

Ok, noted. 





> ![B@8}0FU%LXR%P5A2LIJ@K(3](https://user-images.githubusercontent.com/63439268/180632276-72c4d4bb-4e50-4f3f-99dd-c99a16c1c5c3.png) ![3~%))9JEK1USXSCD$ 8DHH0](https://user-images.githubusercontent.com/63439268/180632278-48460a61-55ac-4e5d-a6c6-78dfc5add399.png)
> 
> when press K

I've removed localization on the commands that use blocks/items/unit names. 



> 1.mp4 
>  2.mp4

Also fixed in e5c09a16b

**Comment by Linnnnnna** [comment]
![a](https://user-images.githubusercontent.com/63439268/184347286-3963d9dd-c07a-4e48-a61b-f3c6272e1ff4.jpg)
